src/adapter/z-stack/adapter/zStackAdapter.ts

```diff
@@ -625,6 +625,22 @@ class ZStackAdapter extends Adapter {
                             noderelation: match.payload.noderelation,
                         });
                     }
+                    // Figure out once if the network address has been changed.
+                    try {
+                        checkedNetworkAddress = true;
+                        const actualNetworkAddress = await this.requestNetworkAddress(ieeeAddr);
```

Comment:
Could you move this logic into a function? I see the same code is duplicated on line 570

Reply:
Yep, for sure. I had started doing that but it will require a bit of cleanup for all the local variables, so I figured I'd wait until I knew if this was going to break things.